utils/near-performance-metrics/src/stats_enabled.rs

```diff
@@ -168,6 +175,34 @@ pub(crate) fn get_entry() -> Arc<Mutex<ThreadStats>> {
     LOCAL_STATS.with(|x| (*x).clone())
 }
 
+#[cfg(all(target_os = "linux", feature = "c_memory_stats"))]
+fn get_c_memory_usage_cur_thread() -> ByteSize {
+    // hack to get memory usage stats for c memory usage per thread
+    // This feature will only work if near is started with environment
+    // LD_PRELOAD=${PWD}/bins/near-c-allocator-proxy.so nearup ...
+    // from https://github.com/near/near-memory-tracker/blob/master/near-dump-analyzer
+    unsafe { ByteSize::b(libc::malloc(usize::MAX - 1) as u64) }
+}
+
+#[cfg(any(not(target_os = "linux"), not(feature = "c_memory_stats")))]
+fn get_c_memory_usage_cur_thread() -> ByteSize {
+    Default::default()
+}
+
+#[cfg(all(target_os = "linux", feature = "c_memory_stats"))]
+fn get_c_memory_usage() -> ByteSize {
+    // hack to get memory usage stats for c memory usage
+    // This feature will only work if near is started with environment
+    // LD_PRELOAD=${PWD}/bins/near-c-allocator-proxy.so nearup ...
+    // from https://github.com/near/near-memory-tracker/blob/master/near-dump-analyzer
+    unsafe { ByteSize::b(libc::malloc(usize::MAX) as u64) }
+}
```

Comment:
Btw, in rust-analyzer we use jemalloc's API for that: https://github.com/rust-analyzer/rust-analyzer/blob/master/crates/profile/src/memory_usage.rs#L27-L31

obv, works only with jemalloc.

In theory, theres' API in glibc for that as well (mallinfo) but in practice it's just broken

Reply:
Yes, there is nothing we can do about it.

Reply:
Also by doing this approach, we can get memory stats per thread, which is nice.